components/device-mgt/io.entgra.device.mgt.core.device.mgt.api/src/main/java/io/entgra/device/mgt/core/device/mgt/api/jaxrs/service/impl/DeviceManagementServiceImpl.java

```
@@ -1739,7 +1739,11 @@ public Response updateOperationStatus(
             DeviceMgtAPIUtils.getDeviceManagementService().updateOperation(device, operation);
 
             if (MDMAppConstants.AndroidConstants.OPCODE_INSTALL_APPLICATION.equals(operation.getCode()) ||
-                    MDMAppConstants.AndroidConstants.OPCODE_UNINSTALL_APPLICATION.equals(operation.getCode())) {
+                    MDMAppConstants.AndroidConstants.OPCODE_UNINSTALL_APPLICATION.equals(operation.getCode()) ||
+                    MDMAppConstants.WindowsConstants.INSTALL_ENTERPRISE_APPLICATION.equals(operation.getCode()) ||
```

This can be a switch case statement, where the default case can be to throw a bad request saying unsupported operation code instead of returning a 200 response currently if there is no matching operation code. 

We can also move the DeviceMgtAPIUtils.getDeviceManagementService().updateOperation() inside this block so both the device operation and subscription status will be updated together for better clarity.